src/CommonAccessors.jl

```diff
@@ -0,0 +1,108 @@
+# Common get and set methods
+
+# NOTE this could be reduced with macros and function generation to even less code.
+
+export getLabel, label
+export getTags, setTags!, tags
+export getTimestamp, setTimestamp!, timestamp
+export getInternalId, interalId
+
+
+const DataLevel0 = Union{VariableDataLevel0, FactorDataLevel0}
+const DataLevel1 = Union{VariableDataLevel1, FactorDataLevel1}
+const DataLevel2 = Union{VariableDataLevel2, FactorDataLevel2}
+
+
+"""
+$SIGNATURES
+
+Return the label for a variable or factor.
+"""
+getLabel(v::DataLevel0) = v.label
+
+"""
+$SIGNATURES
+
+Return the label for a variable or factor.
+
+DEPRECATED label -> getLabel
+"""
+function label(v::DataLevel0)
+  @warn "Deprecated label, use getLabel instead."
+  getLabel(v)
+end
+
+
+"""
+$SIGNATURES
+
+Return the tags for a variable.
+"""
+getTags(v::DataLevel0) = v.tags
+
+
+"""
+$SIGNATURES
+
+Return the tags for a variable.
+
+DEPRECATED, tags -> getTags
+"""
+function tags(v::DataLevel0)
+  @warn "tags deprecated, use getTags instead"
+  getTags
+end
+
+"""
+$SIGNATURES
+
+Set the tags for a factor.
+"""
+setTags!(f::DataLevel0, tags::Vector{Symbol}) = f.tags = tags
+
+
+"""
+$SIGNATURES
+
+Get the timestamp from a DFGFactor object.
+"""
+getTimestamp(v::DataLevel1) = v.timestamp
+
+"""
+$SIGNATURES
+
+Get the timestamp from a DFGFactor object.
+
+DEPRECATED -> use getTimestamp instead.
+"""
+function timestamp(v::DataLevel1)
+    @warn "timestamp deprecated, use getTimestamp instead"
+    getTimestamp(v)
+end
+
+"""
+$SIGNATURES
+
+Set the timestamp of a DFGFactor object.
+"""
+setTimestamp!(v::DataLevel1, ts::DateTime) = v.timestamp = ts
```

Comment:
Also see #182. For me a timestamp is considered to be fixed to a variable and can not be changed. 
It is the same as the label in that sense. If you change the label its not the same variable any more.

Reply:
I'm not so sure about that -- but if there is a strong performance reason then its worth it.  Otherwise you build a factor graph, realize you want to change something small and it becomes a major exercise.  This might be part of #182 for the usecase of changing immutable fields, we would need to create a new variable or factor object and replace the old one in the factor graph.  Will add a comment there.

Reply:
That means, at least for this PR, that summaries will fail on this call. This is a known issue until the immutable/mutable discussion is resolved.